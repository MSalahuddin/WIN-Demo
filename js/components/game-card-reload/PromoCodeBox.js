import React, { useState } from 'react';
import styled from 'styled-components/native';
import PropTypes from 'prop-types';
import {Platform} from 'react-native';
import DottedBoxContainer from '../common/DottedBoxContainer';
import { color, fontFamily } from '../../styles';
import { scale, getWindowWidth, scaleHeight, scaleWidth } from '../../platformUtils';
import { gameCardReloadStrings } from '../../stringConstants';
import { error } from '../../../assets/images';

const Container = styled.View`
  align-items: center;
`;

// full screen minus paddings (2 x 24) and outer box border (16)
const inputBoxWidth = getWindowWidth() - scale(48) - scale(16);
const InputBox = styled.TextInput`
  border-radius: ${scale(8)};
  font-size: ${scale(18)};
  height: ${scaleHeight(48)};
  justify-content: center;
  padding-top: ${Platform.OS ==='android' ?scaleHeight(10) : scaleHeight(3)};
  width: ${inputBoxWidth};
  ${({ backgroundColor }) => backgroundColor && `background-color: ${backgroundColor}`};
  ${({ customFont }) => customFont && `font-family: ${customFont}`};
  ${({ fontColor }) => fontColor && `color: ${fontColor}`};
`;

const Icon = styled.Image`
  height: 20px;
  width: 20px;
`;


const ExclamationMark = styled.View`
  position: absolute;
  right: ${scaleWidth(12)};
  top: ${scaleHeight(12)};
  height: ${scaleHeight(24)};
  width: ${scaleHeight(24)};
`;





const PromoCodeBox = ({ promoCode, setPromoCode, onPromoCodeEnter, isPromoCodeValid, setIsPromoCodeValid }) => {
  const [isFocused, setIsFocused] = useState(false);
  let backgroundColor = isFocused ? color.white : color.darkerWhite;
  let borderColor = isFocused ? color.whitePurple : color.lightGrey;
  let customFont = isFocused ? fontFamily.calibreSemiBold : fontFamily.calibreRegular;
  let dotColor = isFocused ? color.yellow : color.darkerWhite;
  let fontColor = isFocused ? color.lightBlack : color.darkGrey;
  let placeholderText = isFocused ? null : gameCardReloadStrings.enterCouponCode;
  if (isPromoCodeValid === false) {
    backgroundColor = color.darkerWhite;
    borderColor = color.pink;
    customFont = fontFamily.calibreSemiBold;
    dotColor = color.powderPink;
    fontColor = color.warningRed;
    placeholderText = null;
  }

  return (
    <Container>
      <DottedBoxContainer borderWidth={8} borderColor={borderColor} dotColor={dotColor} innerWidth={inputBoxWidth}>
        <InputBox
          allowFontScaling={false}
          testID="promo-code-box-input"
          backgroundColor={backgroundColor}
          textAlign="center"
          placeholder={placeholderText}
          placeholderTextColor={color.darkGrey}
          onChangeText={value => {
            setIsPromoCodeValid(null);
            setPromoCode(value);
          }}
          onFocus={() => setIsFocused(true)}
          onBlur={() => setIsFocused(false)}
          value={!isPromoCodeValid ? promoCode : null}
          customFont={customFont}
          fontColor={fontColor}
          autoCapitalize="none"
          autoCorrect={false}
          onSubmitEditing={onPromoCodeEnter}
        />

        {isPromoCodeValid === false &&
          <ExclamationMark>
            <Icon source={error} resizeMode="contain" />
          </ExclamationMark>
        }
      </DottedBoxContainer>
    </Container>
  );
};

PromoCodeBox.propTypes = {
  promoCode: PropTypes.string,
  setPromoCode: PropTypes.func.isRequired,
  onPromoCodeEnter: PropTypes.func.isRequired,
  isPromoCodeValid: PropTypes.bool,
  setIsPromoCodeValid: PropTypes.func.isRequired,
};

PromoCodeBox.defaultProps = {
  promoCode: '',
  isPromoCodeValid: null
};

export default PromoCodeBox;
