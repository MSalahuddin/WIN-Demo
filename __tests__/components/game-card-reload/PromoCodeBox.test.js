import React from 'react';
import { render } from 'react-native-testing-library';

import PromoCodeBox from '../../../js/components/game-card-reload/PromoCodeBox';

describe('Component PromoCodeBox', () => {
  test('renders correctly', () => {
    const setPromoCode = jest.fn();
    const onPromoCodeEnter = jest.fn();
    const setIsPromoCodeCorrect = jest.fn();
    const setIsPromoCodeValid = jest.fn();

    const output = render(
      <PromoCodeBox
        setPromoCode={setPromoCode}
        onPromoCodeEnter={onPromoCodeEnter}
        setIsPromoCodeCorrect={setIsPromoCodeCorrect}
        setIsPromoCodeValid={setIsPromoCodeValid}
        isPromoCodeCorrect={null}
      />
    );
    expect(output).toMatchSnapshot();
  });
});
