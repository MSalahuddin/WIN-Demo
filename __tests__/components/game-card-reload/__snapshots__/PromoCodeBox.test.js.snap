// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component PromoCodeBox renders correctly 1`] = `
<View
  style={
    Array [
      Object {
        "alignItems": "center",
      },
    ]
  }
>
  <View
    backgroundColor="#B7B7B7"
    borderColor="#B7B7B7"
    borderWidth={8}
    style={
      Array [
        Object {
          "borderRadius": 12,
        },
      ]
    }
  >
    <View
      bdWidth={8}
      isTop={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "flexDirection": "row",
            "height": 8,
            "justifyContent": "space-around",
            "paddingHorizontal": 4,
            "position": "absolute",
            "top": -8,
            "width": 654,
          },
        ]
      }
      width={654}
    >
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
    </View>
    <View
      bdWidth={8}
      isBottom={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "bottom": -8,
            "flexDirection": "row",
            "height": 8,
            "justifyContent": "space-around",
            "paddingHorizontal": 4,
            "position": "absolute",
            "width": 654,
          },
        ]
      }
      width={654}
    >
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
    </View>
    <View
      bdWidth={8}
      isLeft={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "space-around",
            "left": -8,
            "paddingVertical": 4,
            "position": "absolute",
            "width": 8,
          },
        ]
      }
    >
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
    </View>
    <View
      bdWidth={8}
      isRight={true}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "100%",
            "justifyContent": "space-around",
            "paddingVertical": 4,
            "position": "absolute",
            "right": -8,
            "width": 8,
          },
        ]
      }
    >
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
    </View>
    <View
      left={-6}
      size={8}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 16,
            "justifyContent": "center",
            "position": "absolute",
            "width": 16,
          },
        ]
      }
      top={-6}
    >
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
    </View>
    <View
      right={-6}
      size={8}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 16,
            "justifyContent": "center",
            "position": "absolute",
            "width": 16,
          },
        ]
      }
      top={-6}
    >
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
    </View>
    <View
      bottom={-6}
      left={-6}
      size={8}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 16,
            "justifyContent": "center",
            "position": "absolute",
            "width": 16,
          },
        ]
      }
    >
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
    </View>
    <View
      bottom={-6}
      right={-6}
      size={8}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": 16,
            "justifyContent": "center",
            "position": "absolute",
            "width": 16,
          },
        ]
      }
    >
      <View
        dotColor="rgba(249,249,249,0.9)"
        style={
          Array [
            Object {
              "backgroundColor": "rgba(249,249,249,0.9)",
              "borderRadius": 3,
              "height": 6,
              "width": 6,
            },
          ]
        }
      />
    </View>
    <TextInput
      allowFontScaling={false}
      autoCapitalize="none"
      autoCorrect={false}
      backgroundColor="rgba(249,249,249,0.9)"
      customFont="CalibreRegular"
      fontColor="#BFBFBF"
      onBlur={[Function]}
      onChangeText={[Function]}
      onFocus={[Function]}
      onSubmitEditing={[MockFunction]}
      placeholder="Enter your coupon code here..."
      placeholderTextColor="#BFBFBF"
      rejectResponderTermination={true}
      style={
        Array [
          Object {
            "backgroundColor": "rgba(249,249,249,0.9)",
            "borderRadius": 12,
            "color": "#BFBFBF",
            "fontFamily": "CalibreRegular",
            "fontSize": 27,
            "height": 96,
            "justifyContent": "center",
            "paddingTop": 6,
            "width": 654,
          },
        ]
      }
      testID="promo-code-box-input"
      textAlign="center"
      underlineColorAndroid="transparent"
      value=""
    />
  </View>
</View>
`;
